lib/puppet/type/acl/ace.rb

```
@@ -29,14 +29,17 @@ def initialize(permission_hash, provider = nil)
       self.id = permission_hash['id']
       @mask = permission_hash['mask']
       self.rights = permission_hash['rights']
-      self.type = permission_hash['type']
+      self.perm_type = permission_hash['perm_type']
+      if permission_hash['type']
+        self.perm_type = permission_hash['type']
+      end
```

Closer, I would put the deprecation warning here. If both values are set, I would throw an error.